tutorials/integration.md

```diff
@@ -792,26 +642,23 @@ events emitted by the transaction. To get the last caller from the event require
 Putting it all together gives us this function which returns the previous holder of 
 the flag.
 
-[source.javascript]
-```
+```javascript
 const callThroughGsn = async (contract, provider) => {
-	.
-	.
-	.
-	const result = receipt.logs.
-		map(entry => contract.interface.parseLog(entry)).
-		filter(entry => entry != null)[0];
-
-	return result.values['0']
-
+                const transaction = await contract.captureFlag()
+                const receipt = await provider.waitForTransaction(transaction.hash)
```

Comment:
I think you can get the logs by calling await transaction.wait(), without this very cumbersome filtering.